src/VersionTool/Program.cs

```diff
@@ -225,13 +253,324 @@ private static int OnList(CommandOption pathOption)
+
+                    var repo = repos.Where(r => r.Packages.Contains(packageName)).SingleOrDefault();
+                    if (repo != null)
+                    {
+                        // update version
+                        OnUpdateVersion(
+                            new List<string>(new[] { Path.Combine(repo.Path, "src", packageName) }),
+                            new List<string>(),
+                            parsedPackageUpdate[1]);
+
+                        // update entry in packages
+                        packages.Where(p => string.Equals(p.Name, packageName, StringComparison.OrdinalIgnoreCase)).Single().NewVersion = newPackageVersion;
+                    }
+                    else
+                    {
+                        // update dependency
+                        OnUpdateDependency(
+                            new List<string>(new[] { directory }),
+                            new List<string>(),
+                            parsedPackageUpdate[0],
+                            parsedPackageUpdate[1]);
+                    }
+                }
+            }
+
+            // Handle repo updates
+            Console.WriteLine("Update repo");
+            if (updateRepoOption.HasValue())
+            {
+                foreach (var repoUpdate in updateRepoOption.Values)
+                {
+                    var repo = repos.Where(r => string.Equals(r.Name, repoUpdate, StringComparison.OrdinalIgnoreCase)).Single();
+
+                    UpdatePackagesInRepo(repo, directory, packages, rules);
+                }
+            }
+
+            // Cascade updates
+            Console.WriteLine("Update cascade");
+            foreach (var repo in repos.OrderBy(r => r.Order))
+            {
+                if (ContainsModifiedSrcProjectFile(repo.Path))
+                {
+                    UpdatePackagesInRepo(repo, directory, packages, rules);
+                }
+            }
+
+            UpdatePatchConfig(patchConfig, repos);
+            SaveJObject(patchConfigPath, patchConfig);
+
+            return 0;
+        }
+
+        private static void UpdatePackagesInRepo(Repository repo, string directory, IEnumerable<Package> packages, IEnumerable<Rule> rules)
+        {
+            foreach (var package in EnumerateProjects(new[] { Path.Combine(repo.Path, "src") }))
+            {
+                var unUpdatePackage = packages.Where(p => p.Name == package.Name && p.CurrentVersion == p.NewVersion).SingleOrDefault();
+                if (unUpdatePackage != null)
+                {
+                    Console.WriteLine($"{package.Name} is a src project and has not been updated in this patch, updating version.");
+
+                    var rule = GetRuleForPackage(package, rules);
+
+                    if (rule == null)
+                    {
+                        throw new Exception($"No update rule found for package {package.Name} {package.Version}.");
```

Comment:
I would say let's keep rules explicit. I don't want to accidentally update some versions to something unexpected. For example we had a patch where tools were versioned 1.0.0-preview2-final to 1.0.0-preview4-final. In that case, if you didn't find a specific rule it would have been updated to 1.0.1-preview2-final.